docs/howtos/optax_update_guide.rst

```diff
@@ -0,0 +1,281 @@
+learning rate schedule as a parameter for ``learning_rate``.
+
+
+.. |optax.scale_by_schedule()| replace:: ``optax.scale_by_schedule()``
+.. _optax.scale_by_schedule(): https://optax.readthedocs.io/en/latest/api.html#optax.scale_by_schedule
+.. |optax.inject_hyperparams()| replace:: ``optax.inject_hyperparams()``
+.. _optax.inject_hyperparams(): https://optax.readthedocs.io/en/latest/api.html#optax.inject_hyperparams
+
+.. codediff::
+  :title_left: ``flax.optim``
+  :title_right: ``optax``
+
+  def train_step(step, optimizer, batch):
+    grads = jax.grad(loss)(optimizer.target, batch)
+    return step + 1, optimizer.apply_gradient(grads, learning_rate=schedule(step))
+
+  ---
+
+  tx = optax.chain(
+      optax.trace(decay=momentum),
+      # Note that we still want a negative value for scaling the updates!
+      optax.scale_by_schedule(lambda step: -schedule(step)),
+  )
+
+Multiple Optimizers / Updating a Subset of Parameters
+-----------------------------------------------------
+
+In Flax, traversals are used to specify which parameters should be updated by an
+optimizer. And you can combine traversals using
+:py:class:`flax.optim.MultiOptimizer` to apply different optimizers on different
+parameters. The equivalent in Optax is |optax.masked()|_ and |optax.chain()|_.
+
+Note that the example below is using :py:mod:`flax.traverse_util` to create the
+boolean masks required by |optax.masked()|_ - alternatively you could also
+create them manually, or use |optax.multi_transform()|_ that takes a
+multivalent pytree to specify gradient transformations.
+
+Beware that |optax.masked()|_ flattens the pytree internally and the inner
+gradient transformations will only be called with that partial flattened view of
+the params/gradients. This is not a problem usually, but it makes it hard to
+nest multiple levels of masked gradient transformations (because the inner
+masks will expect the mask to be defined in terms of the partial flattened view
+that is not readily available outside the outer mask).
+
+.. |optax.masked()| replace:: ``optax.masked()``
+.. _optax.masked(): https://optax.readthedocs.io/en/latest/api.html#optax.masked
+.. |optax.multi_transform()| replace:: ``optax.multi_transform()``
+.. _optax.multi_transform(): https://optax.readthedocs.io/en/latest/api.html#optax.multi_transform
+
+.. codediff::
+  :title_left: ``flax.optim``
+  :title_right: ``optax``
+
+  kernels = flax.traverse_util.ModelParamTraversal(lambda p, _: 'kernel' in p)
+  biases = flax.traverse_util.ModelParamTraversal(lambda p, _: 'bias' in p)
+
+  kernel_opt = flax.optim.Momentum(learning_rate, momentum)
+  bias_opt = flax.optim.Momentum(learning_rate * 0.1, momentum)
+
+
+  optimizer = flax.optim.MultiOptimizer(
+      (kernels, kernel_opt),
+      (biases, bias_opt)
+  ).create(variables['params'])
+
+  ---
+
+  kernels = flax.traverse_util.ModelParamTraversal(lambda p, _: 'kernel' in p)
```

Comment:
Oh, I'm sorry to have missed this in the first round of review but I'm curious -- why do we need to be using `ModelParamTraversal` at all? Presumably other users of Optax, e.g. those using Haiku, have some other pattern? I'd love to eventually deprecate `ModelParamTraversal` as I don't think it's needed anymore, and if needed replace it with something pretty much like a tree_map that gives you the full path of subkeys rather than just the values.

Reply:
There's plenty of different ways how you can create those masks.

I took `ModelParamTraversal` because it's the most straightforward transition from existing code, but feel free to update the example to use `traverse_utils.flatten_dict()` instead.

Reply:
OK I'll give it a shot.